presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -232,6 +234,7 @@ private SqlQueryScheduler(
         this.stageLinkages = stageLinkages.build();
 
         this.executor = queryExecutor;
+        this.session = session;
```

Comment:
nit: Instead of storing `session`, how about we only store `getMaxConcurrentMaterializations(session)`?